src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -237,30 +240,27 @@ public Path getJarPath(String jarName) {
    */
   public void validate() throws AppEngineException {
     if (sdkPath == null) {
-      throw new AppEngineException("Validation Error: SDK path is null");
```

Comment:
yeah the best thing to do here is to put Preconditions.checkNotNull() in the constructor for each param that is not nullable.